### PR TITLE
Updates on ipyleaflet documentation

### LINKED ISSUE #915: Properly use Sphinx for generating the API docs?
From the docstrings

### PR DESCRIPTION
**Pull request by HaudinFlorence**
use autodoc to replace the attributes and methods tables of the documentation by the autogeneration using docstrings
write the docstrings of velocity.Velocity class
add a page named API reference page for the leaflet module
add changes to homogenize the structure/names of the sections of the different pages

Fix #915

### CODE REVIEW

File: docs/res.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+There is a programmable error in your configuration file:
+
+Traceback (most recent call last):
+  File "/home/florence/mambaforge/envs/ipyleaflet-dev/lib/python3.9/site-packages/sphinx/config.py", line 332, in eval_config_file
+    exec(code, namespace)
+  File "/home/florence/ipyleaflet/docs/source/conf.py", line 29, in <module>
+    html_theme_path = [pydata_sphinx_theme.get_html_theme_path()]
+AttributeError: module 'pydata_sphinx_theme' has no attribute 'get_html_theme_path'
+
+make: *** [Makefile:57: html] Error 2
```
Comment (davidbrochart) [inline docs/res.txt:14]: I guess you didn't mean to commit this file?

File: docs/source/api_reference/map.rst [hunk 2]
```diff
@@ -65,56 +67,9 @@ You can save the ``Map`` and all its layers and controls to an HTML page using t
     The saved file is a static HTML page, so there is no possible interaction with Python anymore. This means that all the Python callbacks you defined (`e.g.` on marker move) cannot be executed. If you want to serve the ``Map`` widget to an HTML page while keeping a Python kernel alive on the server, you might want to look at `Voilà <https://voila.readthedocs.io>`_.
 
 
-Attributes
+Attributes and methods
 ----------
```
Comment (davidbrochart) [inline docs/source/api_reference/map.rst:71]: Don't you have to align the number of `-`?

File: docs/test.wkt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+MULTIPOLYGON(((-71.1031880899493 42.3152774590236,
```
Comment (davidbrochart) [inline docs/test.wkt:1]: Did you intend to commit this file?

File: ipyleaflet/leaflet.py [hunk 4]
```diff
@@ -248,7 +248,7 @@ class Icon(UILayer):
 
     Attributes
     ----------
-    icon_url : string
+    icon_url : string, dafault ""
```
Comment (davidbrochart) [inline ipyleaflet/leaflet.py:251]:
```suggestion
    icon_url : string, default ""
```

File: res.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+make: *** No rule to make target 'html'.  Stop.
```
Comment (davidbrochart) [inline res.txt:1]: I guess you didn't want to commit this file.

**Comment by martinRenou** [comment]
Thanks a lot! This is great.

The ReadTheDocs build fails because it's missing yarn, you should probably remove the `requirements.txt` file and replace it with and `environment.yml` file containing:
```yml

name: ipyleaflet-docs

channels:
  - conda-forge

dependencies:
  - yarn
  - jupyterlab
  - jupyter-packaging
  - jupyter-sphinx
  - scipy
  - netcdf4
  - geopandas
  - xarray
  - branca
  - shapely
  - pip:
    - .
```

**Comment by martinRenou** [comment]
You will also need to change the readthedocs config file according to https://docs.readthedocs.io/en/latest/guides/conda.html#making-builds-faster-with-mamba in order to use mamba

**Review by martinRenou — APPROVED** [review]
This is looking great thanks!

**Pull request merged**